app/src/componentsV2/CodeEditor/components/EditorV2/components/VariablePopOver/components/CreateVariableView.tsx

```diff
@@ -0,0 +1,300 @@
+      return record.id;
+    }
+
+    // If the current record is a request, use its parent collection ID
+    if (record.type === RQAPI.RecordType.API) {
+      return record.collectionId || undefined;
+    }
+
+    return undefined;
+  }, [recordId, getData]);
+
+  const { scopeOptions, defaultScope } = useScopeOptions(collectionId);
+  const { createVariable, status } = useCreateVariable(collectionId);
+
+  const [formData, setFormData] = useState({
+    scope: defaultScope,
+    type: EnvironmentVariableType.String,
+    initialValue: "" as VariableValueType,
+    currentValue: "" as VariableValueType,
+  });
+
+  const [validationError, setValidationError] = useState<string | null>(null);
+
+  // Update default scope when it changes
+  useEffect(() => {
+    setFormData((prev) => ({ ...prev, scope: defaultScope }));
+  }, [defaultScope]);
+
+  const validate = (): boolean => {
+    if (formData.type === EnvironmentVariableType.Number) {
+      const initialValueProvided = formData.initialValue !== "" && formData.initialValue !== undefined;
+      const currentValueProvided = formData.currentValue !== "" && formData.currentValue !== undefined;
+
+      if (initialValueProvided) {
+        const initialNum = Number(formData.initialValue);
+        if (isNaN(initialNum)) {
+          setValidationError("Initial value must be a valid number");
+          return false;
+        }
+      }
+
+      if (currentValueProvided) {
+        const currentNum = Number(formData.currentValue);
+        if (isNaN(currentNum)) {
+          setValidationError("Current value must be a valid number");
+          return false;
+        }
+      }
+    }
+
+    setValidationError(null);
+    return true;
+  };
+
+  const handleTypeChange = useCallback((newType: EnvironmentVariableType) => {
+    setFormData((prev) => {
+      let convertedInitialValue = prev.initialValue;
+      let convertedCurrentValue = prev.currentValue;
+
+      switch (newType) {
+        case EnvironmentVariableType.Boolean:
+          convertedInitialValue = Boolean(convertedInitialValue ?? true);
+          convertedCurrentValue = Boolean(convertedCurrentValue ?? true);
+          break;
+        case EnvironmentVariableType.Number:
+          convertedInitialValue = isNaN(Number(convertedInitialValue)) ? 0 : Number(convertedInitialValue);
+          convertedCurrentValue = isNaN(Number(convertedCurrentValue)) ? 0 : Number(convertedCurrentValue);
+          break;
+        case EnvironmentVariableType.String:
+        case EnvironmentVariableType.Secret:
+          convertedInitialValue = String(convertedInitialValue ?? "");
+          convertedCurrentValue = String(convertedCurrentValue ?? "");
+          break;
+      }
+
+      return {
+        ...prev,
+        type: newType,
+        initialValue: convertedInitialValue,
+        currentValue: convertedCurrentValue,
+      };
+    });
+  }, []);
+
+  const handleSave = async () => {
+    if (!validate()) {
```

Comment:
Validation fails for cross type values, selected number, entered string, still it saves